### DISCUSSION TITLE
Fixing Zoom Styling Issues with Next/Image Component

### DISCUSSION THREAD

**Discussion by Anshtripathi079** [rpearce/react-medium-image-zoom#656] [Q&A]
Hi,

I’m using the `Next/Image` component to render images and applying a zoom functionality to these images. Currently, the zoom effect is not working as expected. Instead of zooming in to a larger size, the image appears to be shrinking. I have some styling issues where the zoomed image seems to be resized incorrectly.

Here’s a visual representation of the problem:

**Current Behavior:**

![Screenshot showing the issue](https://github.com/user-attachments/assets/d0475a7d-ab60-4e2f-bbe0-bb6971b6f6a6)

The zoomed image is smaller than expected:

![Image showing the incorrect zoom effect](https://github.com/user-attachments/assets/a67d24c6-eaf6-45ae-a508-fcaeb1b1ea26)

**Desired Outcome:**

I want to apply these fixed styles to all zoomed images, such as:

```css
top: 50%;
width: 100%;
height: 100%; 
transform: translate(0, -50%) scale(1);
```

Expected Output:

![image](https://github.com/user-attachments/assets/ccf1268b-368e-4cef-a175-3c01e345dea8)

Here's how the current code looks:

```javascript

    let imgComponent = (
            <Image
                src={props.src}
                alt={altText ? altText : 'Image'}
                loading={props.loading}
                height={0}
                width={0}
                sizes="100%"
                className={`h-auto w-full rounded-md border border-solid border-neutral-300 object-contain ${props.className || ''}`}
            />
    );
    // Zoom effect for image if disableZoom is not set
    if (!props.disableZoom) {
        imgComponent = <Zoom>{imgComponent}</Zoom>; 
    }
```

**Comment by rpearce** [comment]
This is the same root issue as https://github.com/rpearce/react-medium-image-zoom/issues/363, where custom transforms and positioning (when zooming) on the image, itself, are not forwarded because transform and position are used to do the zooming.

I'll try to think of how to take this into account—even with just partial support—but no guarantees. The positioning and transforming is the core of how this works, and it's a little complicated.
